streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java

```diff
@@ -69,6 +70,9 @@
                 Optional.empty()
             );
         } catch (final Exception deserializationException) {
+            if (deserializationException instanceof ConfigException) {
```

Comment:
Where does the `ConfigException` come from?

Reply:
IIRC it's deep in `sourceNode.deserializeKey()` but I ran into this in early testing so it may be obsolete. I removed